extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/DefaultTenantConfigResolver.java

```diff
@@ -267,6 +267,9 @@ public Uni<? extends TenantConfigContext> apply(OidcTenantConfig tenantConfig) {
                     if (tenantContext == null) {
                         return tenantConfigBean.createDynamicTenantContext(tenantConfig);
                     } else {
+                        if (tenantContext.oidcConfig() != tenantConfig) {
```

Comment:
Can you please clarify what is happening here

Reply:
If `io.quarkus.oidc.runtime.TenantConfigContext` has `OidcTenantConfig` instance that is different to the one currently used by Quarkus OIDC extension, create copy of `TenantConfigContext` with request-specific `OidcTenantConfig`. This is necessary because Quarkus OIDC extension uses both configs interchangeably and we need consistent behavior.

And I know what you are thinking, why I didn't do this bit separately in a new PR. I don't know how else can I stop using old setters/fields without making this change. In general, I'd expect this change to be correct behavior with/without this PR.

Reply:
>And I know what you are thinking, why I didn't do this bit separately in a new PR

LOL, well, this is what I'm thinking because I'm not sure that if detect such a mismatch it can be effectively ignored by creating a safe copy. This whole state management thing is complex and deserves a discussion on what does it mean to have such a mismatch, should we fail, should we try to adapt. etc

Reply:
> LOL, well, this is what I'm thinking because I'm not sure that if we detect such a mismatch it can be effectively ignored by creating a safe copy.

You do not detect such a mismatch, you allow it to happen.

> This whole state management thing is complex and deserves a discussion on what does it mean to have such a mismatch, should we fail, should we try to adapt. etc

So user will return `instance2`, and you will somewhere use `instance1` stored on the tenant context and elsewhere use `instance2`. Without any plan whatsoever to avoid complexity that I can't see...

Reply:
This change should be no-brainer and you come with `LOL`, I am strongly disagree with your arguments.

Reply:
There is a better way to say it - this change is here because I needed to fix `CodeFlowTest#testAuthenticationCompletionFailedWrongRedirectUri` when `OidcTenantConfigBuilder` created a new instance of `OidcTenantConfig` provider dynamically by a custom `TenantConfigResolver`. I think that was the only test failing, but I am not 100 % sure, because I experienced a lot of failures in past week.

Reply:
OK. I will leave it to you if and when we continue. I am fine just letting someone else having a go into deprecated setters.